sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java

```diff
@@ -44,15 +47,15 @@ public void register(final @NotNull IHub hub, final @NotNull SentryOptions optio
       try {
         Class.forName("androidx.lifecycle.DefaultLifecycleObserver");
         Class.forName("androidx.lifecycle.ProcessLifecycleOwner");
-        watcher =
-            new LifecycleWatcher(
-                hub,
-                this.options.getSessionTrackingIntervalMillis(),
-                this.options.isEnableSessionTracking(),
-                this.options.isEnableAppLifecycleBreadcrumbs());
-        ProcessLifecycleOwner.get().getLifecycle().addObserver(watcher);
-
-        options.getLogger().log(SentryLevel.DEBUG, "AppLifecycleIntegration installed.");
+        if (MainThreadChecker.isMainThread()) {
+          addObserver(hub);
+          options.getLogger().log(SentryLevel.DEBUG, "AppLifecycleIntegration installed.");
+        } else {
+          // some versions of the androidx lifecycle-process require this to be executed on the main
+          // thread.
+          final Handler handler = new Handler(Looper.getMainLooper());
+          handler.post(() -> addObserver(hub));
```

Comment:
that's my suggestion to just not ignore if it's not the main thread.

Reply:
sounds good